t/t2262-job-list-purge.t

```diff
@@ -0,0 +1,181 @@
+#!/bin/sh
+
+test_description='Test flux job list purge inactive'
+
+. $(dirname $0)/sharness.sh
+
+export FLUX_CONF_DIR=$(pwd)
+test_under_flux 4 job
+
+RPC=${FLUX_BUILD_DIR}/t/request/rpc
+
+fj_wait_event() {
+  flux job wait-event --timeout=20 "$@"
+}
+
+get_completed_count() {
+    count=`flux jobs --stats-only 2>&1 | awk '{print \$3}'`
+    echo ${count}
+}
+
+get_failed_count() {
+    count=`flux jobs --stats-only 2>&1 | awk '{print \$5}'`
+    echo ${count}
+}
+
+# Return the expected jobids list in a given state:
+#   "inactive", "completed", "failed"
+#
+state_ids() {
+    for f in "$@"; do
+        cat ${f}.ids
+    done
+}
+
+# Return the expected count of jobs in a given state (See above for list)
+#
+state_count() {
+    state_ids "$@" | wc -l
+}
+
+test_expect_success 'job-list: submit 6 jobs' '
+        #  Create `hostname` and `sleep` jobspec
+        #
+        flux jobspec --format json srun -N1 hostname > hostname.json &&
+        flux jobspec --format json srun -N1 sleep 300 > sleeplong.json &&
+        #
+        # submit jobs that will complete
+        #
+        for i in $(seq 1 4); do \
+                flux job submit hostname.json >> completedids; \
+                fj_wait_event `tail -n 1 completedids` clean ; \
+        done &&
+        #
+        #  Currently all completed ids are also inactive
+        #
+        tac completedids | flux job id > completed.ids &&
+        cp completedids inactiveids &&
+        #
+        #  Run a job that will fail, copy its JOBID to both inactive and
+        #   failed lists.
+        #
+        jobid=`flux mini submit nosuchcommand` &&
+        fj_wait_event $jobid clean &&
+        echo $jobid >> inactiveids &&
+        flux job id $jobid > failedids &&
+        #
+        #  Run a job that will fail, copy its JOBID to both inactive and
+        #   failed lists.
+        #
+        jobid=`flux mini submit --time-limit=0.25 sleeplong.json` &&
+        fj_wait_event $jobid clean &&
+        echo $jobid >> inactiveids &&
+        flux job id $jobid >> failedids &&
+        #
+        #  Submit 8 sleep jobs to fill up resources
+        #
+        for i in $(seq 0 7); do \
+                flux job submit sleeplong.json >> runningids; \
+                fj_wait_event $jobid alloc; \
+        done &&
+        tac runningids | flux job id > running.ids &&
+        #
+        #  Submit a job and cancel it, copy its JOBID to both inactive
+        #    and failed lists
+        #
+        jobid=`flux mini submit canceledjob` &&
```

Comment:
I was at first confused because `canceledjob` is not a command. Why not submit a real command here (maybe give it `--job-name=canceledjob` to aid in debugging)?